infra/gravity/runtest.go

```diff
@@ -0,0 +1,43 @@
+package gravity
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"regexp"
+	"runtime"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+type TestFunc func(ctx context.Context, t *testing.T, config *ProvisionerConfig, payload interface{})
+
+const (
+	Parallel   = true
+	Sequential = false
+)
+
+var reFuncName = regexp.MustCompile(`^([\w\/\.]+)\/([\d\w]+)\.([\d\w]+)$`)
+
+// Run is a handy wrapper around test runner to have a properly assigned tag, state dir, etc.
+func Run(ctx context.Context, t *testing.T, config *ProvisionerConfig, fn TestFunc, payload interface{}, parallel bool) bool {
```

Comment:
I'm not a big fan of `interface{}`. In this case, I'd encapsulate the payload in the `TestFunc` - e.g. instead of relaying it starting in Run, just create a TestFunc as a closure with the necessary parameters.